### PR TITLE
Add changing default device recipe

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by msaroufim — APPROVED** [review]
noice

File: .jenkins/build.sh [hunk 1]
```diff
@@ -29,6 +29,13 @@ pip install -r $DIR/../requirements.txt
 # pip uninstall -y torch torchvision torchaudio torchtext
 # pip install --pre --upgrade -f https://download.pytorch.org/whl/test/cu116/torch_test.html torch torchdata torchvision torchaudio torchtext
 
+# Install 2.0 nightly
+
+pip uninstall -y torch torchvision torchaudio torchtext torchdata
+#pip install --pre --upgrade -f https://download.pytorch.org/whl/test/cu118/torch_test.html torch torchdata torchvision torchaudio torchtext
+pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/test/cu117
+pip3 install torchdata torchtext --index-url https://download.pytorch.org/whl/test/cpu
+
```
Comment (malfet) [inline .jenkins/build.sh:38]:
```suggestion
```

File: .jenkins/validate_tutorials_built.py [hunk 2]
```diff
@@ -51,6 +51,9 @@
     "recipes/Captum_Recipe",
     "flask_rest_api_tutorial",
     "text_to_speech_with_torchaudio",
+    "forward_ad_usage", # remove after #2218 is resolved.
+    "text_sentiment_ngrams_tutorial", # reenable after 2.0 release.
+    "tensorboard_profiler_tutorial" # reenable after 2.0 release.
```
Comment (malfet) [inline .jenkins/validate_tutorials_built.py:56]:
```suggestion
```

Comment (malfet) [inline .jenkins/validate_tutorials_built.py:56]:
```suggestion
```

File: beginner_source/introyt/autogradyt_tutorial.py [hunk 3]
```diff
@@ -334,7 +334,7 @@ def forward(self, x):
     
 print(model.layer2.weight.grad[0][0:10])
 
-optimizer.zero_grad()
+optimizer.zero_grad(set_to_none=False)
```
Comment (malfet) [inline beginner_source/introyt/autogradyt_tutorial.py:337]:
```suggestion
optimizer.zero_grad()
```

**Review by malfet — APPROVED** [review]

**Pull request merged**